### PR TITLE
Improve default point size for binned scatterplot (incomplete)

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: src/compile/scale/range.ts [hunk 1]
```diff
@@ -99,22 +108,45 @@ function parseUnitScaleRange(model: UnitModel) {
...
+
+  if (hasDiscreteDomain(scaleType)) {
+    const range = scaleCmpt && scaleCmpt.get('range');
+    if (range && isVgRangeStep(range) && isNumber(range.step)) {
+      return range.step;
+    }
+    // TODO: support the case without range step
+  } else if (fieldDef && fieldDef.bin) {
+    if (isBinning(fieldDef.bin)) {
+      const binSignal = model.getName(vgField(fieldDef, {suffix: 'bins'}));
```
[the remark this answers is not in the document]
Comment (domoritz) [reply src/compile/scale/range.ts]: Merging two bin operators in the dataflow? Yes, we do have that.

**Comment by kanitw** [comment]
Ok. It seems like we need a signal component that can be revisited during assemble phase (to update signal names that might have been renamed while merging).

Previously, we simply use `SignalRef` for this as we mostly use signal name directly without addition computation.  

However, as we see in this PR, we need to use multiple signals to calculate the bin size range. Thus we need better support for renaming signal names in the assemble phases. 

There are 2 possible approaches:

1. Use `SignalRef` as the signal component.  During assemble, we then call `vega-expression` to parse the expression signal and detect all signal names involved. 
Then for each signal in the expression, we can check if they have been renamed.  

(To provide a more general support, we should revise the `layoutSizeNameMap` to become a more general `signalNameMap` -- as currently we only have the rename map for layout size only.) 


2. Introduce a new `ExpressionSignalComponent` that lists all the signal names that may need to be checked. This won't require us to call Vega expression. However, it might be more error prone as we need to correctly specify dependent signal names. 

I'm leaning towards 1. as it would be a safer programming model for us. Though would incur a bit additional cost in the compilation process, it's probably safer and (more scalable for our own development) than 2. 

@domoritz  Any thoughts?

**Comment by domoritz** [comment]
When we detected that a signal name has changed, do we need to rewrite the expression?

**Comment by kanitw** [comment]
Yep, rename it in the expression

**Comment by domoritz** [comment]
That seems pretty dangerous. What if one name is a subset of the name of another signal?

**Comment by kanitw** [comment]
> What if one name is a subset of the name of another signal?

Not sure why you worry about this since we can write a regular expression that matches whole word only.  Plus, if something got renamed, the only choice we have is to update it to use the new name.  The question is just how we detect something is renamed.

**Comment by domoritz** [comment]
Replaced by #4405

**Pull request closed without merging**